index/scorch/optimize_knn.go

```diff
@@ -36,35 +37,48 @@ func (o *OptimizeVR) Finish() error {
 	// for each VR, populate postings list and iterators
 	// by passing the obtained vector index and getting similar vectors.
 	// defer close index - just once.
-
+	errors := make([]error, len(o.snapshot.segment))
```

Comment:
Don't see why this is necessary - you can just declare this as ..

```
var errors []error
```

And then where ever you encounter an err, append to that list (which will alloc).

Replace line 79-81 below with ..
```
if len(errors) > 0 {
    return errors[0]
}
```